### ISSUE TITLE
[getter-return] Extend ESLint's getter return to better fit TypeScript

### ISSUE THREAD

**Issue by noftaly** [typescript-eslint/typescript-eslint#2897]
Hi!

I'd like to suggest extending the [ESLint's `getter-return` rule](https://eslint.org/docs/rules/getter-return) to better fit TypeScript.
I ran into this quite specific case where, in my getter, I `switch` a value which is an `enum`, and in my `case`s, I checked all the enum values, so I don't need any default case. But the getter-return rule asks me to add a default value...
Example code:
```ts
enum Animals {
    Dog,
    Cat,
}
class MyClass {
    animal: Animals;

    get animal() {
        switch (this.animal) {
            case Animals.Dog: return 'blahblah1';
            case Animals.Cat: return 'blahblah2';
            // All cases have been tested, this line will never be reached, yet getter-return asks me to put a default value :)
        }
    }
    set animal(value: Animals) {
        this.animal = value;
    }
}
```

**Additional Info**
I don't have any specific Fail/Pass cases to show as this is just an extension.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.11.0` |
| `@typescript-eslint/parser`        | `4.11.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.16.0` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
I don't think it's worth the effort of enhancing this rule with type information.

TypeScript already 100% handles this:

```ts
enum Animals {
    Dog,
    Cat,
}

class Foo {
  get bar() {} // Error: A 'get' accessor must return a value. (2378)
  
  _animal: Animals = Animals.Dog;

  get baz() { // Error: Not all code paths return a value. (7030)
    switch(this._animal) {
      case Animals.Dog: return 'dog';
    }
  }

  get bam(): string { // Error: Function lacks ending return statement and return type does not include 'undefined'.(2366)
    switch(this._animal) {
      case Animals.Dog: return 'dog';
    }
  } 

  get foo(): string | undefined { // Error: Not all code paths return a value.(7030)
    switch(this._animal) {
      case Animals.Dog: return 'dog';
    }
  } 
}
```
[repl](https://www.typescriptlang.org/play?#code/KYOwrgtgBAgiCWECGAbAzlA3gKCnqAIgPYDmANLvgMJIAuFAvttgMYpJoYBiRRWlJYLSgAjJACcAFAEosDKAHoFUAKLjxRcQC5YUAOSDaeqEhYtgnTVAhg0w8ULDiQJqADdUYYADookgEwAzADsABzSlJQA+kgIyCg6cIioGAC8sHEp3sQkANzMeIaiSABeMliKymoa2lAAckTCqChQLEQAJsBQAA50ABYYDrROLkjunj5+wQAMgdMR+FBoAO7wtCx9krR98GjeMZkosjiLeCwcXUnxezk6QyP67aR6+YtMeEwCQsUQMjp24ngIBIFSUqnUmh0XDAIBYtHgRBc7BYAGsMKB2kCQfdnEtaHRgBBQE0QO0oDiXLQAJ7dLpPCxQECNKBAthgTr6GGdABmQOA7T03gCgQAbCKFvgVmsNlsdnsDskjvxTq0LhlFTdSHdHLi9E8SC9KB9KPIClAitzeH88YDgVAAD5QLnAXkgfmgqoQ2oNJooFptDm9baDHWjcYoLxCmZzCV4KXrTbbXb7WKK45G-DnNCXQ6akja4a6-WGt4mqDYBhAA)

So using the `getter-return` lint rule is entirely superfluous and duplicates the TS errors.
[If you use our `eslint-recommended` config, we disable the rule for you](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/eslint-recommended.ts#L12).
